### ISSUE TITLE
Description flag doesn't allow newlines(?) in 0.23.0

### ISSUE THREAD

**Issue by zerodahero** [lucassabreu/clockify-cli#126]
It seems that the description option no longer works with newlines in the string.

```bash
> clockify-cli version
Version: 0.23.0, Commit: 7c3e9c4d907b22e018bad8c7811699c55de46821, Build At: 2021-09-17T00:08:29Z
```

```bash
> clockify-cli edit current --description "Line one
Line two"
EOF
> echo $?
1
```

No other error message, and nothing is updated in clockify.

Please let me know if you would like any more information from me to help with this. I can try and take a look and maybe submit a PR, but wanted to raise the issue, first--in case I'm missing something. Thanks!

**Comment by lucassabreu** [comment]
i didnt know it worked with multi-line... can you run it again with the `--debug` flag and copy-paste here @zerodahero ? it may contain your token so, remove it before pasting.

**Comment by zerodahero** [comment]
Oh, actually, scratch that. That's the only subcommand (`edit`) that I use newlines for (so apologies for not testing more thoroughly), but it looks like editing just the description doesn't work (single line or otherwise).

```bash
> clockify-cli edit current --description="anything"
EOF
```

Debug:
There's actually quite a bit of sensitive info in here, so I can scrub the rest of it if that would be helpful, the main line that catches my eye (last one) is:
```
DEBUG 2021/09/17 12:14:34 url: https://api.clockify.me/api/v1/workspaces/<workspaceID>/time-entries/<timeEntryID>?hydrated=true, status: 403, body: ""
```
All other statuses are 200

**Comment by lucassabreu** [comment]
if you run a `clockify-cli log in-progress` the time entry id that shows up is the same that would be at "<timeEntryID>"?

it fails to edit other properties, like tags, task or project?

the 403 really confuses me

**Comment by zerodahero** [comment]
Ah! `clockify-cli log in-progress` looks normal, but in comparing the time entry ID to the url segment in the debug, they don't match.

**Comment by lucassabreu** [comment]
it is probably a problem with finding the `current` time entry on edit subcommand.

i wont be able to look at it right now, but if you feel like trying is probably a problem here:
https://github.com/lucassabreu/clockify-cli/blob/7c3e9c4d907b22e018bad8c7811699c55de46821/cmd/common.go#L544-L559

i believe it is returning the wrong time entry, when you set as "current" it should use the current or the most recent one, the is not retrieving the "in-progress" one from the looks of it

**Comment by lucassabreu** [comment]
if you use the time-entry id manually it should work

**Comment by zerodahero** [comment]
I got a quick change to poke around with that. It looks like it may be more API related with Clockify than code-related. Seems like the old API (that `GetRecentTimeEntries` hits) doesn't return them in a sorted order. It looks like despite the limit of 1, it's returning the last completed time entry _and_ the current one.

The new API looks like it returns both a different structure and slightly different data, so it may be a slightly larger change to go that route. My guess would be that the short-term solution (using the old api) would be to check for a list.len > 1 and return either last or current based on the `running: <bool>` value from the API?

**Comment by lucassabreu** [comment]
i think is a good patch for now, can you send a pr with it @zerodahero ?

**Comment by lucassabreu** [comment]
i explored the new api and it returns than sorted so i will be able to change to use it, but it will a hydrated=false call to the api to work, so it will be a lot bigger.

**Comment by zerodahero** [comment]
Ok, opened up a PR for that (let me know if there's any improvements I can make on it--I'm still fairly new to Golang). It addresses the issue with the editing the current entry, but there is still a 403 on getting the modified record (from the new API).

**Comment by lucassabreu** [comment]
patched on [v0.23.1](https://github.com/lucassabreu/clockify-cli/releases/tag/v0.23.1)

**Comment by lucassabreu** [comment]
now that i read this, even if you use the real id instead of the aliases it fails with 403?

> Ok, opened up a PR for that (let me know if there's any improvements I can make on it--I'm still fairly new to Golang). It addresses the issue with the editing the current entry, but there is still a 403 on getting the modified record (from the new API).

**Comment by zerodahero** [comment]
Yes, I got in touch with Clockify support. They kinda confirmed that the other bug addressed in #127 is a bug/regression in their old API, but they aren't going to do anything about it since it's deprecated.

The 403s still occur for me on directly fetching a specific time entry. It happens when directly hitting the API, so it doesn't seem to be an issue with this tool. I'm still chatting with Clockify support on this, but as far as a "code issue" goes, this can probably be closed. I'll post any updates in case that helps others in the future.

**Comment by lucassabreu** [comment]
lets keep it open then, so if anyone else see this problem they can follow this issue.

**Comment by zerodahero** [comment]
Quick update: they've escalated the issue to their dev team to investigate further.

In the meantime, the cli works as expected, with the exception that it isn't able to display the modified entry after modification. So any "edit" call for me is successful in the modification (on Clockify), but the result of the command will show the 403.

I'll update here with any more info from Clockify.

**Comment by lucassabreu** [comment]
@zerodahero thank you for updating me.
